gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/storage/application/ApplicationStorageServiceInMemory.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.rest.api.storage.application;
+
+import io.gravitee.rest.api.model.BaseApplicationEntity;
+import io.gravitee.rest.api.service.common.ExecutionContext;
+import io.gravitee.rest.api.service.exceptions.ApplicationNotFoundException;
+import io.gravitee.rest.api.storage.InMemoryStorageService;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class ApplicationStorageServiceInMemory implements ApplicationStorageService, InMemoryStorageService<BaseApplicationEntity> {
+
+    private final List<BaseApplicationEntity> storage = new ArrayList<>();
+
+    @Override
+    public BaseApplicationEntity findById(ExecutionContext executionContext, String applicationId) {
+        return storage
+            .stream()
+            .filter(application -> applicationId.equals(application.getId()))
+            .findFirst()
+            .orElseThrow(() -> new ApplicationNotFoundException(applicationId));
```

Comment:
Why not filter by environment like in `ApplicationStorageServiceImpl` ? If this logic has to be implemented at this level then it should probably be done by all different implementations right ?

Reply:
Because the `ApplicationEntity` does not have `environmentId` attribute 🙈

Reply:
:aa: